### DISCUSSION TITLE
You can now checkin other users (API only)

### DISCUSSION THREAD

**Discussion by MrKrisKrisu** [Traewelling/traewelling#2806] [Announcements]
Hey @Traewelling/api-consumer,

in [Release v1.79](https://github.com/Traewelling/traewelling/releases/tag/v1.79) we added the possibility to check in other users via the API. Users must allow this beforehand. There is a new property in the UserSettings for this purpose. All these features are currently API-only and not built into the traewelling.de user interface.

### Settings

[Added a new UserSettings](https://traewelling.de/api/documentation#/Settings/updateProfileSettings) to specify who is allowed to checking in for a user (forbidden, friends, list).

Possible values:

- `forbidden` - default value
- `friends` = Users who follow each other can check in
- `list` = "trusted users" can check in. see below.

### New Attribute `with` on [Checkin-Endpoint](https://traewelling.de/api/documentation#/Checkin/createCheckin)

You can just add up to 10 user IDs for check-in in addition to the actual user.

![image](https://github.com/Traewelling/traewelling/assets/4103693/c411f381-898c-42cd-b27b-614f893b1704)

### New [Endpoints](https://traewelling.de/api/documentation#/User/trustedUserIndex) to manage "trusted users"
Trusted Users are only required, if User allow a specific list of trusted users to check in for them.
![image](https://github.com/Traewelling/traewelling/assets/4103693/036cc61d-390d-42fd-92fb-7a1ff4238d95)

### New NotificationType `YouHaveBeenCheckedIn`

```json
{
   "status": 1,
   "checkin": {
      "line": "ICE 123",
      "origin": "Karlsruhe Hbf",
      "destination": "Hannover Hbf"
   },
   "user": {
      "id": 1,
      "username": "Getrud123"
      "name": "Gertrud"
   }
}
```

**Comment by marhei** [comment]
Thank you for implementing this long awaited feature.

There is only one problem: for a proper implementation of this feature, an endpoint for getting all users who trust me is missing.

**Reply by jheubuch** [reply]
That was exactly my thought when I had a look at this yesterday :D

**Reply by MrKrisKrisu** [reply]
There will be a new endpoint available for this in the next release. :)

https://github.com/Traewelling/traewelling/pull/2823

![image](https://github.com/user-attachments/assets/6b2bdf99-affb-494a-8cc2-2484969c9b01)

**Comment by marhei** [comment]
In addition: It would be helpful to also return the current state of the `friendCheckin` setting on the `/settings/profile` endpoint.

**Reply by MrKrisKrisu** [reply]
Will be available with the next release. :)

https://github.com/Traewelling/traewelling/pull/2822

**Comment by marhei** [comment]
The `/user/self/trusted` endpoint returns a paginated list, the `/user/self/trusted-by` does not. For more consistency between both endpoints I would propose to make the `/user/self/trusted` non-paginated as well.

**Reply by MrKrisKrisu** [reply]
> [!CAUTION]
> possible breaking change, if you've already implemented trusted users in your application

The endpoint `/user/{self/userId}/trusted` will be NOT-paginated from the next release. As this endpoint is new, I think this change can be made without problems. The structure itself won't change. Only the pagination is no longer available and all users will be printed with one request.

https://github.com/Traewelling/traewelling/pull/2831

**Comment by marhei** [comment]
Small bug on the `/user/self/trusted-by` endpoint: It will always return users who have added you to their trusted users, even if they have `friendCheckin` set to `forbidden` and you can't check them in.

**Reply by MrKrisKrisu** [reply]
fixed in https://github.com/Traewelling/traewelling/pull/2856
